File: 2-ad-hoc/1266.py
while(x:=int(input())):
    array=[]
    cont,index,e,tot=0,0,0,0
    for i in input().split():
        tot+=int(cont/2) if index and i=="1" else 0
        cont += 1 if i=="0"else -cont
        index=e+1 if i=="1" else index
        e+=1
        array.append(i)
    cont=0
    for i in range(x+1):
        analysis=array[index+i] if i<x-index else array[i-x+index]
        if analysis=='1':break
        cont+=1
    tot+=cont/2
    print(int(tot))

File: 2-ad-hoc/1266.c
#include<stdio.h>
//
int main(){
        for(;;){
                int length,temp,initCont=0,cont=0,tot=0,init=0;
                scanf("%i",&length);
                if(!length)
                        break;
                int array[length];
                for(int i=0;i<length;++i){
                        scanf("%i",&temp);
                        array[i]=temp;
                        temp?init=i:init;
                }
                for(int i=0,analysis;i<length+1;++i){
                        analysis=i<length-init?array[init+i]:array[i-length+init];
                        analysis?tot+=cont/2:tot;
                        !analysis?++cont:analysis?cont=0:cont;
                }
                tot+=cont/2;
                printf("%i\n",tot);
        }
        return 0;
}
